thirdparty/cmdconfig/commands/cmdtree/cmdtree.go

```diff
@@ -83,107 +48,24 @@ func (r *TreeRunner) runE(c *cobra.Command, args []string) error {
 		input = &kio.ByteReader{Reader: c.InOrStdin()}
 	} else {
 		root = filepath.Clean(args[0])
-		input = kio.LocalPackageReader{PackagePath: args[0]}
-	}
-
-	var fields []kio.TreeWriterField
-	for _, field := range r.fields {
-		path, err := runner.ParseFieldPath(field)
-		if err != nil {
-			return err
-		}
-		fields = append(fields, newField(path...))
-	}
-
-	if r.name || (r.all && !c.Flag("name").Changed) {
-		fields = append(fields,
-			newField("spec", "containers", "[name=.*]", "name"),
-			newField("spec", "template", "spec", "containers", "[name=.*]", "name"),
-		)
-	}
-	if r.images || (r.all && !c.Flag("image").Changed) {
-		fields = append(fields,
-			newField("spec", "containers", "[name=.*]", "image"),
-			newField("spec", "template", "spec", "containers", "[name=.*]", "image"),
-		)
-	}
-
-	if r.cmd || (r.all && !c.Flag("command").Changed) {
-		fields = append(fields,
-			newField("spec", "containers", "[name=.*]", "command"),
-			newField("spec", "template", "spec", "containers", "[name=.*]", "command"),
-		)
-	}
-	if r.args || (r.all && !c.Flag("args").Changed) {
-		fields = append(fields,
-			newField("spec", "containers", "[name=.*]", "args"),
-			newField("spec", "template", "spec", "containers", "[name=.*]", "args"),
-		)
-	}
-	if r.env || (r.all && !c.Flag("env").Changed) {
-		fields = append(fields,
-			newField("spec", "containers", "[name=.*]", "env"),
-			newField("spec", "template", "spec", "containers", "[name=.*]", "env"),
-		)
+		input = kio.LocalPackageReader{PackagePath: args[0], MatchFilesGlob: r.getMatchFilesGlob()}
 	}
 
-	if r.replicas || (r.all && !c.Flag("replicas").Changed) {
-		fields = append(fields,
-			newField("spec", "replicas"),
-		)
-	}
-	if r.resources || (r.all && !c.Flag("resources").Changed) {
-		fields = append(fields,
-			newField("spec", "containers", "[name=.*]", "resources"),
-			newField("spec", "template", "spec", "containers", "[name=.*]", "resources"),
-		)
-	}
-	if r.ports || (r.all && !c.Flag("ports").Changed) {
-		fields = append(fields,
-			newField("spec", "containers", "[name=.*]", "ports"),
-			newField("spec", "template", "spec", "containers", "[name=.*]", "ports"),
-			newField("spec", "ports"),
-		)
-	}
-
-	// show reconcilers in tree
 	fltrs := []kio.Filter{&filters.IsLocalConfig{
-		IncludeLocalConfig:    r.includeLocal,
-		ExcludeNonLocalConfig: r.excludeNonLocal,
+		IncludeLocalConfig: true,
 	}}
 
 	return runner.HandleError(c, kio.Pipeline{
 		Inputs:  []kio.Reader{input},
 		Filters: fltrs,
-		Outputs: []kio.Writer{kio.TreeWriter{
+		Outputs: []kio.Writer{TreeWriter{
 			Root:            root,
 			Writer:          c.OutOrStdout(),
-			Fields:          fields,
-			Structure:       kio.TreeStructure(r.structure),
 			OpenAPIFileName: kptfilev1alpha2.KptFileName,
```

Comment:
I found `OpenAPIFileName` to be misleading here. I am guessing this was to enable parameterization of pkg file. We can probably remove this parameter and use the `Kptfile` directly ?

Reply:
Good point. Done.